src/libAtomVM/opcodesswitch.h

```diff
@@ -1793,7 +1793,12 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
                 DECODE_COMPACT_TERM(timeout, code, i, next_off, next_off)
 
                 #ifdef IMPL_EXECUTE_LOOP
-                    if (!term_is_integer(timeout) && UNLIKELY(timeout != INFINITY_ATOM)) {
+                    if (term_is_any_integer(timeout)) {
+                        avm_int64_t t = term_maybe_unbox_int64(timeout);
+                        if (t < 0 || t > 4294967295L) {
```

Comment:
Do we want to limit timers to 32 bits? I wonder if they are 64 in feature/smp.

Reply:
The current docs say that 32 only bit integers are supported:

https://www.erlang.org/doc/reference_manual/expressions.html#receive

I think I had read that the OTP team was going to change that, but I don't see that in the docs (yet).

Reply:
I didn't realize the test was also run with BEAM. But then it should fail with OTP25! We can fix that when we merge the PR for OTP25 support.

Reply:
Right, but it does beg the question, what should _we_ do.  I could see us supporting OTP25 behavior (with the caveat that we don’t support arbitrary-length integers — yet), and just remove the test for this condition in the code and in the unit test.  Unlike the stacktrace question, the OTP team has made the change already, so there would be good reason for us to adopt it.

Reply:
I modified the code to allow 64-bit timeouts.